File: Php/core/FormFields/TimezoneFieldModel.php
<?php
namespace core\FormFields{

	class TimezoneFieldModel extends SelectFieldModel{

		public function __construct($fieldName, $fieldId, $selectedValue, $timezones, $firstItemText = null){
			$items = $this->getSelectItems($timezones);
			parent::__construct($fieldName, $fieldId, $selectedValue, $items, $firstItemText);
		}

		private function getSelectItems($timezones){
			$items = array();
			if($timezones != null){
				foreach($timezones as $timezone){
					$items[] = new SelectFieldItem($timezone);
				}
			}
			return $items;
		}

	}

}

File: Web/Models/FormModels/TimezoneFieldModel.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Web.Models.FormModels{

	public class TimezoneFieldModel : SelectFieldModel{

		public TimezoneFieldModel(string fieldName, string fieldId, string selectedValue, IEnumerable<TimeZoneInfo> timezones, string firstItemText = null)
            : base(fieldName, fieldId, selectedValue, GetSelectItems(timezones), firstItemText)
        {
		}

        private static IEnumerable<SelectFieldItem> GetSelectItems(IEnumerable<TimeZoneInfo> timezones)
        {
			var items = new List<SelectFieldItem>();
			if(timezones != null){
			    items.AddRange(timezones.Select(timezone => new SelectFieldItem(timezone.DisplayName, timezone.Id)));
			}
			return items;
		}

	}

}